src/main/java/kr/syeyoung/dungeonsguide/features/impl/dungeon/FeatureDungeonMap.java

```diff
@@ -267,15 +269,37 @@ private void updateMapTexture(byte[] colors, MapProcessor mapProcessor, List<Dun
         this.mapTexture.updateDynamicTexture();
     }
 
+    final Pattern pattern = Pattern.compile("\\*[a-zA-Z0-9_]{2,16}\\*", Pattern.MULTILINE);
+
+
+
     private void renderHeads(MapProcessor mapProcessor, MapData mapData, float scale, float postScale, float partialTicks) {
         List<NetworkPlayerInfo> list = field_175252_a.sortedCopy(Minecraft.getMinecraft().thePlayer.sendQueue.getPlayerInfoMap());
         if (list.size() < 40) return;
         for (int i = 1; i < 20; i++) {
             NetworkPlayerInfo networkPlayerInfo = list.get(i);
             String name = networkPlayerInfo.getDisplayName() != null ? networkPlayerInfo.getDisplayName().getFormattedText() : ScorePlayerTeam.formatPlayerName(networkPlayerInfo.getPlayerTeam(), networkPlayerInfo.getGameProfile().getName());
+
             if (name.trim().equals("§r") || name.startsWith("§r ")) continue;
-            String actual = TextUtils.stripColor(name).trim().split(" ")[0];
-            EntityPlayer entityplayer = Minecraft.getMinecraft().theWorld.getPlayerEntityByName(actual);
+
+            name = TextUtils.stripColor(name);
+
+            name = name.replaceAll(" ", "*");
+
+
+            Matcher matcher = pattern.matcher(name);
+
+            if(!matcher.find()) continue;
+
+            name = matcher.group(0);
+
+            name = name.substring(0, name.length() - 1);
+
+            name = name.substring(1);
+
+
+            EntityPlayer entityplayer = Minecraft.getMinecraft().theWorld.getPlayerEntityByName(name);
+
             if (TextUtils.stripColor(name).endsWith("(DEAD)") && entityplayer != Minecraft.getMinecraft().thePlayer) {
```

Comment:
Look at this!!
DG checks if player's dead
because well, hypixel likes to put invisible players everywhere!!!

Please re-implement dead player check using tab list.

Reply:
wait wdym im doing stuff exactly as your code, just changed the part where it would cut the name out of player tab text bc it was outputting the "[20]" instead of "username". cuz hypixel started putting lvl in tab.

Reply:
ohhh
SORRY IM STUPID